src/client/localplayer.cpp

```diff
@@ -492,6 +496,12 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 	// Prevent sliding on the ground when jump speed is 0
 	m_can_jump = m_can_jump && jumpspeed != 0.0f;
 
+	// Prevent jump if in node/object
+	if (m_can_jump && could_sneak) {
+		if (collisionCheckIntersection(env, m_client, m_collisionbox, initial_position))
```

Comment:
Isn't this already calculated in L398 - or are those two positions considerably different? Couldn't you keep `sneak_can_jump = false` in L407 in that case?

Reply:
The different position argument is used in those locations. So these calls are not equivalent.